### ISSUE TITLE
Add test coverage for remote server

### ISSUE THREAD

**Issue by daluu** [daluu/phrrs#2]
Use this discussion as starting point:

http://groups.google.com/group/robotframework-devel/browse_thread/thread/3fc989c191300a6b

Base tests off RF acceptance tests for the Python remote server. And expand as needed to test for PHP to <XML-RPC> to  Robot Framework data type conversions.

Ensure that the server passes most of the tests, and fix what doesn't pass, where possible.

**Comment by daluu** [comment]
With the PR, good start on coverage, adequate for now. 

I would prefer to have a normalized tests organization of the following:

* server specific implementation unit test suite (ideally, these wouldn't need to "start up" a remote server, just calling the internal methods, but not required)
* Robot Framework (RF) compatibility test suite, this would likely start up a remote server instance to run a set of (standard) RF tests against. These tests are agnostic to implementation and server specific features.

Good reference examples:

* Python

 * implementation specific unit tests: https://github.com/robotframework/PythonRemoteServer/tree/master/test/utest

 * RF compatibility test suite: https://github.com/robotframework/PythonRemoteServer/tree/master/test/atest

* Java

 * implementation specific unit tests: https://github.com/ombre42/jrobotremoteserver/tree/master/src/test/java/org/robotframework/remoteserver

 * RF compatibility test suite: https://github.com/ombre42/jrobotremoteserver/tree/master/atest

There may already be some overlap of coverage in the phrrs tests, but I'd like to normalize the organization and naming of the tests to match the reference examples. Easier to compare and benchmark with the other remote server implementations that way. Will take some time to review and correlate phrrs tests with the examples noted above and work out reorg/differences.

@jplambert you're welcome to help with what I just mentioned, but you don't have to change your existing tests if you don't want to.

**Comment by jplambert** [comment]
@daluu I agree -- I was not aware of such test suites. Might be easier to implement the RF compatibility test suite first, and then to clean-up the existing tests.
